lib/puppet/configurer.rb

```diff
@@ -402,6 +403,15 @@ def save_last_run_summary(report)
     Puppet.log_exception(detail, "Could not save last run local report: #{detail}")
   end
 
+  def save_last_run_report(report)
+    mode = Puppet.settings.setting(:lastrunreport).mode
+    Puppet::Util.replace_file(Puppet[:lastrunreport], mode) do |fh|
+      fh.print YAML.dump(report)
+    end
+  rescue => detail
+    Puppet.log_execption(detail, "Could not save last run local report: #{detail}")
```

Comment:
Thanks @joshcooper, updated. That also pointed out this case wasn't sufficiently tested - test added.